pkg/addon/agent/controller/token.go

```diff
@@ -375,3 +404,49 @@ func exceedThreshold(now metav1.Time, expiring metav1.Time, lastRefreshTimestamp
 	threshold := lastRefreshTimestamp.Add(expiring.Sub(lastRefreshTimestamp.Time) / 5 * 4)
 	return now.Time.After(threshold), threshold
 }
+
+// getNamespaceFromToken extracts namespace from JWT token payload
+func getNamespaceFromToken(token string) (string, error) {
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/authentication/serviceaccount/util.go#L61 I think you can use this func instead.

Reply:
The namespace could be found in 3 fields of the payload of token:
1. can get namespace from the `kubernetes.io`.`namespace ` directly. 
2. can get namespace from the `sub` string.
3. can get namespace from  the `"kubernetes.io/serviceaccount/namespace"` filed in old k8s version.
the function `MatchesUsername` can only check the namespace in the `sub` filed of the payload of token, but cannot cover all the cases.

Reply:
But I think that is good enough.

Reply:
done. changed to only check the namespace and name in the sub field of payload using `MatchesUsername ` .